gnd/src/main/java/com/google/android/gnd/persistence/geojson/GeoJsonTile.java

```diff
@@ -53,34 +52,22 @@
    * <p>Interior rings, which describe holes in the polygon, are ignored.
    */
   GeoJsonTile(JSONObject jsonObject) {
-    this.extent = new GeoJsonExtent(new GeoJsonGeometry(jsonObject));
-    this.id = Optional.of(jsonObject.optString(ID_KEY));
-    this.url =
-        Optional.ofNullable(jsonObject.optJSONObject(PROPERTIES_KEY))
-            .map(j -> j.optString(URL_KEY));
+    this.json = jsonObject;
   }
 
-  public Optional<LatLngBounds> getBounds() {
-    return this.extent.getBounds();
-  }
-
-  public Optional<LatLng[]> getCoordinates() {
-    return this.extent.getCoordinates();
+  private ImmutableList<LatLng> getVertices() {
+    return new GeoJsonExtent(new GeoJsonGeometry(json)).getVertices();
   }
 
   public Optional<String> getId() {
-    return this.id;
+    return Optional.of(json.optString(ID_KEY));
```

Comment:
This will return Optional.of("") if it doesn't exists. We should either model missing as `null`, `Optional.empty()`, or `""`. Any of those alternatives are fine, but all these will make edge cases harder to catch.

Reply:
Gotcha, returning Optional.empty on...well, empty values.